feat: Introduce a separate parsing_method rule (#244)
mozilla/eslint-plugin-no-unsanitized: docs/rules/parsing_method.md

# parsing_method

The _parsing_method_ rule in _eslint-plugin-no-unsanitized_ perform basic security

> checks for function calls that parse strings into new Document instances. The idea

Reviewer: ```suggestion
checks for function calls that parse strings into new Document or DocumentFragment instances. The idea
```

> of these checks is to allow developers to opt-in/opt-out detecting use of these methods,

Reviewer: ```suggestion
of these checks is to allow developers to opt-in/opt-out of detecting use of these methods,
```

# parsing_method

> The _parsing_method_ rule in _eslint-plugin-no-unsanitized_ performs basic security
> checks for function calls that parse strings into new Document or DocumentFragment
> instances. The idea of these checks is to allow developers to opt-in/opt-out of detecting
> use of these methods, separately from the `method` rule which is reporting violation
> as errors by default.

Reviewer: ```suggestion
The _parsing_method_ rule in _eslint-plugin-no-unsanitized_ performs basic security
checks for function calls that parse strings into new Document or DocumentFragment
instances. It is close to impossible to automatically check whether the resulting
Document is actually used insecurely, so this offers a control at the parse step.
Therefore, these checks allow developers to opt-in/opt-out of detecting
use of these methods, separately from the `method` and `property` rule which is
reporting violation as errors by default.
```
